pkg/controller/volume/attachdetach/attach_detach_controller.go

```diff
@@ -293,6 +294,23 @@ func (adc *attachDetachController) nodeAdd(obj interface{}) {
 	// the attached volumes field. This function ensures that we sync with
 	// the actual status.
 	adc.actualStateOfWorld.SetNodeStatusUpdateNeeded(nodeName)
+
+	if _, exists := node.Annotations[volumehelper.ControllerManagedAttachAnnotation]; exists {
+		// Newly added node is managed by controller, ensure all pods are up to date.
+		adc.processPodsForNode(nodeName)
+	}
+}
+
+func (adc *attachDetachController) processPodsForNode(nodeName types.NodeName) {
+	pods, err := adc.podLister.List(labels.Everything())
```

Comment:
Yeah we'd need a node index to do this efficiently, or use a work queue w/ pod processing retries. Considering #42033 also processes every single pod every 3 mins is it a big deal in comparison? I wasn't sure.

The main purpose for me is to get a fix we can use in 1.6/1.5. I didn't think #42033 was a candidate for backporting, or was even going to stick around in its current form. I thought it was more of a backup, experimental solution. Is it okay to have to wait for the reconciler to fix things when new nodes get added?

If we just want #42033, that's okay with me, as long as we can get a fix in 1.6/1.5 as well, since this is a major problem in those releases.